### PR TITLE
Documentation: Documenting Codes in the Scripts Folder

### LINKED ISSUE #57: Documentation: Documenting Codes in the Scripts Folder
The code in the Scripts folder needs documentation to improve its understanding and facilitate collaboration with other developers.

image

**Tasks:**

1. Add XML comments to methods, classes and public variables.

- Example:
```csharp 
/// <summary> 
/// Controls the movement of the player.
/// </summary> 
/// <param name=“speed”>Speed of the movement.</param> 
void MovePlayer(float speed) { // Implementation here }
```

2. Document the general functionality of each script:

- What the script does.
- Where and how it is used.

### PR DESCRIPTION
**Pull request by josephpdf**
### 📝 Summary 
- Commets all files .cs
- Closes #57 
### 🔄 Changes Made
- Add comments on all files in the Scripts folder
### 🔧 Tests Results
Captura de pantalla 2024-11-22 114110

### CODE REVIEW

File: Unity/Assets/Scripts/CharacterSelectionScript/LoadCharacter.cs [hunk 1]
```diff
@@ -3,27 +3,46 @@
...
+    /// </summary>
+    void Start()
     {
-        // Obt�n el �ndice del personaje seleccionado guardado en PlayerPrefs
-        int selectedCharacter = PlayerPrefs.GetInt("selectedCharacter", 0); // Usa 0 como valor predeterminado
+        int selectedCharacter = PlayerPrefs.GetInt("selectedCharacter", 0); 
         if (selectedCharacter < 0 || selectedCharacter >= characterPrefab.Length)
         {
-            Debug.LogError("�ndice de personaje seleccionado est� fuera de rango. Verifica los prefabs asignados en el inspector.");
+            Debug.LogError("�ndice de personaje seleccionado est� fuera de rango. Verifica los prefabs asignados en el inspector.");
```
Comment (Josue19-08) [inline Unity/Assets/Scripts/CharacterSelectionScript/LoadCharacter.cs:39]: Could you please translate this part into English, please.

File: Unity/Assets/Scripts/DroneMovement/DroneMovement.cs [hunk 2]
```diff
@@ -1,30 +1,52 @@
...
-        
+        // Update drone position
         transform.position = new Vector3(x, y, z);
 
-        
+        // Apply additional rotation around the right axis
         transform.Rotate(Vector3.right, rotationSpeed * Time.deltaTime);
 
-        
+        // Make the drone face its movement direction
```
Comment (Josue19-08) [inline Unity/Assets/Scripts/DroneMovement/DroneMovement.cs:49]: You could remove the comments and add a single comment inside the <summary>

File: Unity/Assets/Scripts/WeaponsScripts/AtivationWeapons.cs [hunk 3]
```diff
@@ -2,23 +2,39 @@
...
+    /// When triggered, activates the corresponding weapon in the player's inventory
+    /// and destroys the pickup object.
+    /// </summary>
+    /// <param name="other">The Collider that entered this object's trigger zone</param>
     private void OnTriggerEnter(Collider other)
     {
         if (other.CompareTag("Player"))
         {
-            // Intentar obtener CollectionWeapons en el momento de la colisi�n
+            // Intentar obtener CollectionWeapons en el momento de la colisi�n
```
Comment (Josue19-08) [inline Unity/Assets/Scripts/WeaponsScripts/AtivationWeapons.cs:31]: Remove this comment

File: Unity/Assets/Scripts/WeaponsScripts/AtivationWeapons.cs [hunk 4]
```diff
@@ -2,23 +2,39 @@
...
             CollectionWeapons collectionWeapons = other.GetComponentInChildren<CollectionWeapons>();
 
             if (collectionWeapons == null)
             {
                 Debug.LogError("CollectionWeapons component not found on the Player or its children.");
-                return; // Salir del m�todo si CollectionWeapons es nulo
+                return; // Salir del m�todo si CollectionWeapons es nulo
```
Comment (Josue19-08) [inline Unity/Assets/Scripts/WeaponsScripts/AtivationWeapons.cs:37]: Remove this comment

File: Unity/Assets/Scripts/Minimap/MinimapController.cs [hunk 5]
```diff
@@ -1,16 +1,28 @@
...
     private void LateUpdate()
     {
-        if (player == null) return; // Aseg�rate de que el jugador est� asignado
+        if (player == null) return; // Ensure player reference is assigned
 
-        // Actualizar la posici�n de la c�mara para que siga al jugador
+        // Update camera position to follow player while maintaining its height
         Vector3 newPosition = player.position;
-        newPosition.y = transform.position.y; // Mantener la altura configurada en el Inspector
+        newPosition.y = transform.position.y; // Keep height set in Inspector
```
Comment (Josue19-08) [inline Unity/Assets/Scripts/Minimap/MinimapController.cs:25]: You could remove the comments and add a single comment inside the summary

File: Unity/Assets/Scripts/RegistrySystem/ShowNameOtherScene.cs [hunk 6]
```diff
@@ -2,25 +2,38 @@
...
         if (SceneManager.GetActiveScene().name == "DemoSelectionCharacter")
         {
-            // Mostrar "Welcome, [Nombre]" solo en la escena de selección de personaje
+            // Display "Welcome, [Name]" format in character selection scene
             playerNameTetx.text = "Welcome, " + playerName;
         }
         else
         {
-            // Mostrar solo el nombre en otras escenas
+            // Display only the name in other scenes
```
Comment (Josue19-08) [inline Unity/Assets/Scripts/RegistrySystem/ShowNameOtherScene.cs:36]: You could remove the comments and add a single comment inside the summary

File: Unity/Assets/Scripts/WeaponsScripts/CollectionWeapons.cs [hunk 7]
```diff
@@ -2,11 +2,24 @@
...
 
-    // Method to activate a specific sword
+    /// <summary>
+    /// Activates a specific weapon and deactivates all others.
+    /// Used when switching between different weapons in the game.
+    /// </summary>
+    /// <param name="number">The index of the weapon to activate (0-based index)</param>
     public void ActivationWeapon(int number)
     {
         // Validate that the index is within the weapon range
```
Comment (Josue19-08) [inline Unity/Assets/Scripts/WeaponsScripts/CollectionWeapons.cs:25]: You could remove the comments and add a single comment inside the summary

Comment (Josue19-08) [inline Unity/Assets/Scripts/WeaponsScripts/CollectionWeapons.cs:25]: You could remove the comments and add a single comment inside the summary

Comment (Josue19-08) [inline Unity/Assets/Scripts/WeaponsScripts/CollectionWeapons.cs:25]: You could remove the comments and add a single comment inside the summary

Comment (josephpdf) [reply Unity/Assets/Scripts/WeaponsScripts/CollectionWeapons.cs:25]: Okay, I deleted all comments and only I left the comments inside the summary

**Review by Josue19-08 — APPROVED** [review]
Great. Looks good for me.

**Review by CristopherAguilar10 — APPROVED** [review]

**Comment by CristopherAguilar10** [comment]
Yes, all good

**Pull request merged**